pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -578,6 +582,16 @@ def newpkgcb(pkgId, name, arch):
                                                                       filelists_xml_path,
                                                                       other_xml_path)
 
+                        # remove user defined package types before parse
+                        if self.skip:
```

Comment:
So instead of doing this, you just want

```
if self.skip_srpms:
    packages = {pkgId: pkg for pkgId, pkg in packages.items() if pkg.arch != 'src'}
```